Modules/posixmodule.c

```diff
@@ -5088,384 +5088,358 @@ os__path_splitroot_impl(PyObject *module, path_t *path)
 }
 
 
-/*[clinic input]
-os._path_isdir
-
-    s: 'O'
+#define PY_IFREG  1 // Regular file
+#define PY_IFDIR  2 // Directory
+#define PY_IFLNK  4 // Symlink
+#define PY_IFMNT  8 // Mount Point (junction)
+#define PY_IFLRP 16 // Link Reparse Point (name-surrogate, symlink, junction)
+#define PY_IFRRP 32 // Regular Reparse Point
 
-Return true if the pathname refers to an existing directory.
+static BOOL
+_testInfo(DWORD attributes, DWORD reparseTag, int testedType)
+{
+    if (testedType == PY_IFREG) {
+        return attributes && !(attributes & FILE_ATTRIBUTE_DIRECTORY);
+    }
+    if (testedType == PY_IFDIR) {
+        return attributes & FILE_ATTRIBUTE_DIRECTORY;
+    }
+    if (attributes & FILE_ATTRIBUTE_REPARSE_POINT) {
+        switch (testedType) {
```

Comment:
Also consider flipping this switch with the earlier if statements, so the entire body is `switch (testedType) { ... }` and then the conditions below here all test `attributes & FILE_ATTRIBUTE_REPARSE_POINT`.